### PR TITLE
XENG-7690 Allow targeting a stage to build for multi-stage builds

### PR DESCRIPTION
**Pull request by mheidbrink**
### What does this PR do?
Allows you to stop at a specific build stage in multi-stage builds.
See https://docs.docker.com/build/building/multi-stage/#stop-at-a-specific-build-stage

### What issues does this PR fix or reference?
XENG-7690

### Merge requirements satisfied?
- [x] I have updated the documentation or no documentation changes are required.
- [x] I have added tests to cover my changes.
- [x] I have updated the base version in ``setup.py`` (if appropriate).

### CODE REVIEW

File: tests/test-files/test-multi-stage.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+steps:
+  test:
+    build:
+      no-cache: true
+      dockerfile: |
+        FROM {{ DOCKER_REGISTRY }}/busybox:latest AS stage1
+        RUN echo "test" > /test-file.txt
+        FROM {{ DOCKER_REGISTRY }}/busybox:latest AS stage2
+      target: stage1
```
Comment (bluesliverx) [inline tests/test-files/test-multi-stage.yaml:9]: This seems an insufficient test to me, what if you write another file in `stage2` and then use the something like this for the run:

```
run:
  cmds:
  - test -f /test-file1.txt
  - ! test -f /test-file2.txt
```

Comment (bluesliverx) [reply tests/test-files/test-multi-stage.yaml:9]: and obviously change the name of the file written in `stage1` for my suggestion

File: buildrunner/docker/multiplatform_image_builder.py [hunk 2]
```diff
@@ -353,6 +359,7 @@ def build_multiple_images(
         platforms: List[str],
         path: str = ".",
         file: str = "Dockerfile",
+        target: str = None,
```
Comment (bluesliverx) [inline buildrunner/docker/multiplatform_image_builder.py:362]: This should be `Optional[str]`

File: tests/test_multiplatform.py [hunk 3]
```diff
@@ -475,6 +475,7 @@ def test_build_multiple_builds(
             cache_from=None,
             cache_to=None,
             pull=False,
+            target=None,
```
Comment (bluesliverx) [inline tests/test_multiplatform.py:478]: These references should not be needed if it's already the default, right?

Comment (mheidbrink) [reply tests/test_multiplatform.py:478]: That's what I thought, but the test failes if it's not there.  I guess the test is explicitly checking the parameter count that's provided too or something.

Comment (bluesliverx) [reply tests/test_multiplatform.py:478]: Oh, I should have expanded this, I thought this was the method call to the func from the test method, but it's a mocked method, so yeah, the value would be set to none.

**Review by bluesliverx — APPROVED** [review]

**Pull request merged**